tests/tensor/rewriting/test_linalg.py

```diff
@@ -306,3 +306,39 @@ def test_invalid_batched_a(self):
             ref_fn(test_a, test_b),
             rtol=1e-7 if config.floatX == "float64" else 1e-5,
         )
+
+
+@pytest.mark.skipif(
+    config.mode == "FAST_COMPILE",
+    reason="inplace rewrites disabled when mode is FAST_COMPILE",
+)
+def test_local_inplace_cholesky():
+    X = matrix("X")
+    L = cholesky(X, overwrite_a=False, lower=True)
+    f = function([pytensor.In(X, mutable=True)], L)
```

Comment:
I did it in the test because I'm lazy. The rewrite will never apply to a root node and I didn't want to make and check intermediate variables in the test. 

For PyMC, I'm not sure where in a typical logp graph a cholesky decomposition gets applied. As long as it's not directly applied to a random variable after its sampled we should see some benefits.

Reply:
> The rewrite will never apply to a root node and I didn't want to make and check intermediate variables in the test.

Ahh gotcha, that makes sense.  Yah agree this is the simplest test then.